dataset/include/scipp/dataset/map_view.h

```diff
@@ -203,16 +195,10 @@ template <class Base, class Access> class MutableView : public Base {
   MutableView() = default;
   MutableView(const Access &access, typename Base::holder_type &&items,
               const detail::slice_list &slices);
-  /// Constructor for internal use (slicing and holding const view in mutable
-  /// view)
   MutableView(const Access &access, Base &&base);
 
-  /// Return a view to the coordinate for given dimension.
   typename Base::mapped_type::view_type
-  operator[](const typename Base::key_type key) const {
-    scipp::expect::contains(*this, key);
-    return make_slice(Base::items().at(key), Base::slices());
-  }
+  operator[](const typename Base::key_type key) const;
```

Comment:
I don't expect `Base::key_type` to be large or expensive to copy, but shouldn't this be `const &` anyway?

Reply:
Good catch! It may actually be a string in case of masks.